core/trino-main/src/main/java/io/trino/execution/SqlQueryExecution.java

```diff
@@ -475,7 +475,7 @@ private PlanRoot planQuery()
             return doPlanQuery();
         }
         catch (StackOverflowError e) {
-            throw new TrinoException(NOT_SUPPORTED, "statement is too large (stack overflow during analysis)", e);
```

Comment:
This changes the classification of the error from a user error to an internal error. Generally, a stack overflow is caused by query trees that are deeper than the parser/analyzer/planner support, hence the current classification as a user error of type "unsupported".

Reply:
The case we're seeing is due to an infinite loop by the planner. On subsequent retry, the query succeeds. I can keep it as a USER_ERROR if you think that's better, but the main thing we need is a different, more specific error code.

Reply:
I've updated the PR to continue to classify this issue as a USER_ERROR, but with a different error code.